mewarpx/medocker/docker_shared/run_warpx.sh

```diff
@@ -35,8 +25,13 @@ function pingterminatewarning {
             TOKEN=`curl -s -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 30"`
         # 200 means marked for termination
         elif [[ "$HTTP_CODE" -eq 200 ]] ; then
+            # Use the WarpX signal handling functionality to checkpoint the
+            # simulation. Note that we use the checkpointing signal rather than
+            # the break signal (with graceful exiting) otherwise the "attempts"
+            # functionality on AWS Batch does not work as intended.
```

Comment:
This suggests a slight expansion of the set of actions available in upstream WarpX in response to signals - a graceful exit with a non-zero exit code to not confuse the job scheduler into thinking the job finished just before being killed would be good.